### PR TITLE
DataFrame.apply for functions passed as string fix

### LINKED ISSUE #1656: DataFrame.apply fails with function passed as string and added function parameters
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 18.04
- **Modin version** (`modin.__version__`): 0.7.3+153.g4f95b41
- **Python version**: 3.8.3
- **Code we can use to reproduce**:
```
import pandas
import modin.pandas as pd
import numpy as np

rows_number = 20

data = {"col1": np.random.randint(0, 100, rows_number), "col2": np.random.randint(0, 100, rows_number)}
level_0 = np.random.choice([0, 1], rows_number)
level_1 = np.random.choice([2, 3], rows_number)
index = pd.MultiIndex.from_arrays([level_0, level_1])

func_kwargs = {"level": 0}
df_modin = pd.DataFrame(data, index=index)
df_pandas = pandas.DataFrame(data, index=index)

pandas_output = df_pandas.apply("count", **func_kwargs)
print("pandas_output:\n", pandas_output)

modin_output = df_modin.apply("count", **func_kwargs)
print("modin_output:\n", modin_output)
```

### Describe the problem
Reproducer raises exception:
Traceback (most recent call last):
  File "apply_reproducer.py", line 19, in <module>
    modin_output = df_modin.apply("count", **func_kwargs)
  File "/modin/modin/pandas/dataframe.py", line 375, in apply
    if axis == 0 and result.name == self.index[0]:
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

### PR DESCRIPTION
**Pull request by amyskov**
Signed-off-by: Alexander Myskov <alexander.myskov@intel.com>

## What do these changes do?

- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #1656 
- [x] tests added and passing

### CODE REVIEW

File: modin/pandas/test/test_dataframe.py [hunk 1]
```diff
@@ -1727,6 +1727,29 @@ def test_apply(self, request, data, func, axis):
...
+
+        try:
+            pandas_result = pandas_df.apply(func, **func_kwargs)
+
+        except Exception as e:
+            with pytest.raises(type(e)):
+                modin_df.apply(func, **func_kwargs)
+        else:
+            modin_result = modin_df.apply(func, **func_kwargs)
+            df_equals(modin_result, pandas_result)
```
Comment (dchigarev) [inline modin/pandas/test/test_dataframe.py:1751]: there is a function at [`modin/pandas/test/utils.py`](https://github.com/amyskov/modin/blob/aplly_str_func/modin/pandas/test/utils.py#L557) that could simplify this
```suggestion
        eval_general(
            pd.DataFrame(data, index=index)
            pandas.DataFrame(data, index=index)
            lambda df, *args, **kwargs: df.apply(func, *args, **kwargs),
            **func_kwargs,
        ) 
```

**Comment by dchigarev** [comment]
Could you also add at `query_compiler::apply` dispatching to `apply_text_func_elementwise` if passed `func` is `str`?

**Comment by amyskov** [comment]
@dchigarev, `query_compiler::apply` can process text functions now.

**Comment by dchigarev** [comment]
> @dchigarev, `query_compiler::apply` can process text functions now.

Thanks a lot!

**Pull request merged**